user-service/src/main/java/com/rzheng/userservice/controller/UserController.java

```diff
@@ -27,4 +27,9 @@ public ResponseEntity<User> addUser(@RequestBody User user) {
         User savedUser = userService.addUser(user);
         return ResponseEntity.ok(savedUser);
     }
+
+    @GetMapping("/user/check-email")
```

Comment:
`/api/v1/users/user/check-email`. The extra `/user` seems redundant. You can remove the `/user` and the endpoint path looks like `/api/v1/users/check-email`.